src/patterns/InPlaceFilters/FilterButton.tsx

```diff
@@ -0,0 +1,204 @@
+        handleClick(filtersListState.map((selected, i) => (i === index ? !selected : false)));
+      }
+    };
+
+    const { totalCount, filterCount } = countActiveFilters(filters, buttonType);
+    const buttonLabel =
+      `${filterButtonLabel === 'Sort' ? 'Sort By' : filterButtonLabel}` + (filterCount > 0 ? ` (${filterCount})` : '');
+
+    // Main Filter button (shows all filters in a Drawer)
+    if (buttonType === 'Filter' && filters) {
+      return (
+        <>
+          <FilterButtonDisplay
+            ref={ref}
+            className={className}
+            isButtonSelected={isButtonSelected}
+            filterCount={filterCount}
+            totalCount={totalCount}
+            buttonType={buttonType}
+            filterButtonLabel={filterButtonLabel}
+            buttonLabel={buttonLabel}
+            id={id}
+            handleFilterButtonClick={handleFilterButtonClick}
+            isMobileDropdown={false}
+            ariaLabel={ariaLabels.button || 'Filter button'}
+          />
+          <Drawer
+            isOpen={isButtonSelected}
+            drawerOpenSide="left"
+            onClose={handleFilterButtonClick}
+            className={`${px}-filter-drawer`}
+            aria-label={ariaLabels.drawer || 'Filter drawer'}
+          >
+            <FilterMenu className={`${px}-filter-drawer-menu`}>
+              {filters.map((filter: FilterType) => (
+                <Filter key={filter.label} name={filter.label}>
+                  <FilterHeader heading={filter.label} />
+                  {Array.from(filter.filterDimensions).map((value: FilterDimension) => (
+                    <FilterInput
+                      id={value.label}
+                      key={value.label}
+                      labelText={value.label}
+                      onChange={(e) => handleFilterSelection?.(e, filter.label)}
+                      type={filter.type as 'checkbox' | 'radio'}
+                      disabled={value?.disabled}
+                      name={value.label}
+                      checked={value.active}
+                    />
+                  ))}
+                </Filter>
+              ))}
+            </FilterMenu>
+            <div
+              className={classnames(
+                `${px}-filter-dropdown__buttons-wrap`,
+                `${px}-filter-dropdown__buttons-wrap--drawer`,
+              )}
+            >
+              <Button
+                className={`${px}-filter-dropdown__button`}
+                variant={ButtonVariants.secondary}
+                onClick={() => clearFilterUpdate?.('all')}
+              >
+                Clear All
+              </Button>
+              <Button className={`${px}-filter-dropdown__button`} onClick={() => handleFilterUpdate?.(false)}>
+                Show {resultsCount} lots
+              </Button>
+            </div>
+          </Drawer>
+        </>
+      );
+    }
+
+    // Other filter buttons: Popover for desktop, Drawer for mobile
+    return (
```

Comment:
it feels a bit like the "Main Filter" button and "Sub Filter" buttons should be two different components entirely